Sources/generate-swift-syntax-builder/gyb_syntax_support/ExprNodes.py

```diff
@@ -618,13 +618,13 @@
     Node('KeyPathExpr', name_for_diagnostics='key path', kind='Expr',
          children=[
              Child('Backslash', kind='BackslashToken'),
-             Child('RootExpr', kind='Expr', is_optional=True,
+             Child('RootExpr', kind='Expr', name_for_diagnostics='root', is_optional=True,
```

Comment:
I haven't gone through all of these (do you want me to?) but should we be careful to use the swift.org language here? Ie. `type name` and `key path components`?

Also, are child names for optional nodes useful? Would they ever actually be used?

Reply:
> I haven't gone through all of these (do you want me to?) but should we be careful to use the swift.org language here? Ie. `type name` and `key path components`?

Honestly, I didn’t put too much effort into this (still took me 30-60 minutes to create these annotations). If you’ve got an idea how to create better names in a principled way, I’d be open to suggestions. Also, we can always update them later, so I don’t think it has to be blocking this PR.

My idea was that we update these as we discover diagnostics that we don’t like…

> Also, are child names for optional nodes useful? Would they ever actually be used?

They aren’t being used right now but they don’t hurt either. And maybe we’ll come up with a use case later.